packages/peregrine/lib/talons/Logo/useLogo.js

```diff
@@ -0,0 +1,53 @@
+import { useEffect } from 'react';
+import { useQuery } from '@apollo/react-hooks';
+
+/**
+ *
+ * @param query for magento theme logo
+ * @param defaultSrc for logo path
+ * @param defaultWidth for logo width
+ * @param defaultHeight for logo height
+ * @param defaultAlt for logo alt text
+ */
+export const useLogo = props => {
+    const {
+        query,
+        defaultSrc,
+        defaultWidth,
+        defaultHeight,
+        defaultAlt
+    } = props;
+    const { error, data } = useQuery(query);
+
+    useEffect(() => {
+        if (error) {
+            console.log('Error fetching logo data.');
+        }
+    }, [error]);
+
+    let logoSrc;
+
+    if (data && data.storeConfig && data.storeConfig.header_logo_src) {
+        logoSrc =
+            (data &&
+                data.storeConfig &&
+                data.storeConfig.secure_base_media_url) +
+            'logo/' +
+            (data && data.storeConfig && data.storeConfig.header_logo_src);
+    } else {
+        logoSrc = defaultSrc;
```

Comment:
QA found that for a fraction of a second the default logo is displayed while data is fetched. To avoid this we could make it so that we only render if we are done loading. You could add a condition like `if (loading) { logoSrc = null }`.